### PR TITLE
feat: get requests route

### PR DESCRIPTION
**Pull request by Jaszkowic**

### CODE REVIEW

**Review by aeschi — APPROVED** [review]
LGTM as far as I can tell

File: src/lib/routes/load-user-request-route.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+				"/:requestId",
+				{
+					schema: {
+						response: getUserRequestSchema,
+					},
+				},
+				async (request, reply) => {
+					const { requestId } = request.params as { requestId: string };
+
+					const { data, error } = await supabase
```
Comment (ff6347) [inline src/lib/routes/load-user-request-route.ts:70]: Why not handle the error and just check for missing data?

File: src/lib/routes/load-user-request-route.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+									.from("processed_documents")
+									.select("*")
+									.eq("id", matchingDocumentReference.processed_document_id)
+									.single<ProcessedDocument>();
+
+								if (processedDocumentError) {
+									throw new Error("Error fetching processed document");
+								}
+
+								const processedDocumentChunks = await Promise.all(
```
Comment (ff6347) [inline src/lib/routes/load-user-request-route.ts:113]: No error handling here.

File: src/tests/load-user-request.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import anyTest, { TestFn } from "ava";
+import { buildTestServer } from "./util/test-server.js";
+import { FastifyInstance, InjectOptions } from "fastify";
+import { mockServer } from "../mock/node.js";
```
Comment (ff6347) [inline src/tests/load-user-request.test.ts:4]: Are you using msw?

**Review by ff6347 — COMMENTED** [review]
Hi. @Jaszkowic I know I'm late to the party. These are the things  I noticed. I did not get into the logic for now.

File: src/lib/routes/load-user-request-route.ts (hunk 1 above)
Comment (Jaszkowic) [reply src/lib/routes/load-user-request-route.ts:70]: Error handling can definitely be improved here, yes. I just missed handling the error here, iterating on error handling is already a todo ticket.

Comment (ff6347) [reply src/lib/routes/load-user-request-route.ts:70]: Working on that ticket

File: src/lib/build-server.ts [hunk 4]
```diff
@@ -75,6 +76,7 @@ export async function buildServer({
 		OPENAI_MODEL,
 	);
 	registerGenerateAnswerRoute(fastify, OPENAI_MODEL, OPENAI_KEY);
+	registerLoadUserRequestRoute(fastify);
```
[the remark this answers is not in the document]
Comment (ff6347) [reply src/lib/build-server.ts:79]: I also want them to be in their own files but I dislike passing the fastify instance around. I fear sideeffects. My approach would be 

in build-server.ts

```ts
import {routeHandler} from "./routes/handler-function.ts"
await fastify.register(routeHandler)
```

routes/hander-function.ts
```ts
export function routeHandler(request: FastifyRequest, reply: FastifyReply){
//...
}
```

So there is no room to do things to the server in one of the route/files...
Might be nitpicking

[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/lib/build-server.ts:79]: ah I understand, yes that sounds reasonable. feel free to change it accordingly, or put it in the backlog.

**Pull request merged**